build error=unused-variable

##### Brief Description

OS: Linux openSUSE Tumbleweed
qTox version: 1.14.0
Commit hash:  5b120c1

Reproducible: Always

##### Steps to reproduce

1. cmake
```
cmake . '-GUnix Makefiles' -DCMAKE_INSTALL_PREFIX:PATH=/usr -DINCLUDE_INSTALL_DIR:PATH=/usr/include -DLIB_INSTALL_DIR:PATH=/usr/lib64 -DSYSCONF_INSTALL_DIR:PATH=/etc -DSHARE_INSTALL_P
REFIX:PATH=/usr/share -DCMAKE_INSTALL_LIBDIR:PATH=/usr/lib64 -DCMAKE_BUILD_TYPE=RelWithDebInfo '-DCMAKE_C_FLAGS=-fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwi
nd-tables -fstack-clash-protection -g -DNDEBUG' '-DCMAKE_CXX_FLAGS=-fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -g -DNDEBU
G' '-DCMAKE_Fortran_FLAGS=-fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -g -DNDEBUG' '-DCMAKE_EXE_LINKER_FLAGS=-Wl,--as-nee
ded -Wl,--no-undefined -Wl,-z,now' -DCMAKE_MODULE_LINKER_FLAGS=-Wl,--as-needed '-DCMAKE_SHARED_LINKER_FLAGS=-Wl,--as-needed -Wl,--no-undefined -Wl,-z,now' -DLIB_SUFFIX=64 -DCMAKE_SKIP_RPATH:BOOL=ON -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DBUILD_SHAR
ED_LIBS:BOOL=ON -DBUILD_STATIC_LIBS:BOOL=OFF -DCMAKE_COLOR_MAKEFILE:BOOL=OFF -DCMAKE_INSTALL_DO_STRIP:BOOL=OFF -DCMAKE_MODULES_INSTALL_DIR=/usr/share/cmake/Modules
```
2. make

##### Observed Behavior
```
[  118s] /home/abuild/rpmbuild/BUILD/qtox-1.14.0/src/widget/friendwidget.cpp: In member function 'void FriendWidget::onContextMenuCalled(QContextMenuEvent*)':
[  118s] /home/abuild/rpmbuild/BUILD/qtox-1.14.0/src/widget/friendwidget.cpp:144:10: error: unused variable 'friendList' [-Werror=unused-variable]
[  118s]      auto friendList = qobject_cast<FriendListWidget*>(w->parentWidget());
[  118s]           ^~~~~~~~~~
```

--- a/src/core/core.cpp
+++ b/src/core/core.cpp
@@ -713,6 +713,10 @@ bool parseConferenceSendMessageError(TOX_ERR_CONFERENCE_SEND_MESSAGE error)
     case TOX_ERR_CONFERENCE_SEND_MESSAGE_TOO_LONG:
         qCritical() << "Message too long";
         return false;
+
+    default:
+        qCritical() << "Unknown TOX_ERR_CONFERENCE_SEND_MESSAGE error";
+        return false;
     }
 }
 

--- a/src/widget/friendwidget.cpp
+++ b/src/widget/friendwidget.cpp
@@ -139,10 +139,6 @@ void FriendWidget::onContextMenuCalled(QContextMenuEvent* event)
 
     const auto& s = Settings::getInstance();
     const auto circleId = s.getFriendCircleID(frnd->getPublicKey());
-    auto circleWidget = CircleWidget::getFromID(circleId);
-    auto w = circleWidget ? circleWidget : static_cast<QWidget*>(this);
-    auto friendList = qobject_cast<FriendListWidget*>(w->parentWidget());
-
     auto circleMenu = menu.addMenu(tr("Move to circle...",
                                         "Menu to move a friend into a different circle"));
 

